### DISCUSSION TITLE
Containers created with `AddContainer(..).WithLifetime(..)`

### DISCUSSION THREAD

**Discussion by raffaeler** [microsoft/aspire#6924] [Q&A]
I have a solution where the container is added with something like this:
```csharp
var vectordb = builder.AddContainer("vectorDB", "qdrant/qdrant")
    .WithEndpoint(6333, 6333, "http", "dashboard")
    .WithEndpoint(6334, 6334, null, "grpc")
    .WithBindMount(
        @"H:\VMs\ContainerData\qdrant\storage",
        "/qdrant/storage")
    .WithBindMount(
        @"H:\VMs\ContainerData\qdrant\config\production.yaml",
        "/qdrant/config/production.yaml")
    .WithBindMount(
        @"H:\VMs\ContainerData\qdrant\snapshots",
        "/qdrant/snapshots")
    .WithLifetime(ContainerLifetime.Persistent);
```

I love the `ContainerLifetime.Persistent` as it saves me a lot of time between runs.
Anyway, the container is only usable when the solution is running: as soon as you shut down the application the container is still running but the ports are "removed" and the `qdrant` dashboard cannot be reached anymore.

From Docker Destkop, when the Aspire solution is stopped I see the following:
![image](https://github.com/user-attachments/assets/39058d1c-5391-4fb9-837a-7d978e64ee21)

- The `vibrant_joloit` container was created by myself, it's obviously stopped but you can clearly see the port mappings being persisted
- The `vectorDB-db4a9ca4` was created by Aspire, it is still running because of the `ContainerLifetime.Persistent` but has no port mappings and cannot be reached

If I run the Aspire solution again, the `qdrant` dashboard can be reached again.

Is there a way for Aspire to persist the port and volume settings persisted so that it can be always reached?

**Comment by davidfowl** [comment]
Are you trying to use persistent containers as an easy way to setup a container that is always running on your machine? There might be a mismatch on the expected use case.

**Reply by raffaeler** [reply]
Nope. You still need to verify the entries in the database or doing some maintenance, backup, etc. when the Aspire solution is not running.
I do not expect needing to run the entire Aspire solution (8+ processes) to just open the db dashboard and delete a collection. Some maintenance operations may fail when the solution is running.

Also, there are times where you may need to run a project alone, without all the other projects. But if you don't run Aspire, you can't access the database.

The only possible alternative is temporary comment out `AddContainer` and remembering to manually run the container first, but this is not very practical because you also have to keep in sync the endponts and volumes so that they are available in the Aspire manifest.

**Comment by krisnashaypp** [comment]
Would be great if this were something you can opt in to. I'm running into a similar situation where I have to do some work on a python project that exists outside Aspire but uses the same Kafka broker.

This would make the Apphost truly more like a docker compose as well and make it an easier sell I believe.

**Reply by davidfowl** [reply]
It's a really strange workflow IMO, run the application, then close it, then use the containers? It makes sense in docker compose land because you run compose up and down explicitly. The workflow here feels a bit different.

That said, we're working on making it possible to connect to containers that are persistent https://github.com/dotnet/aspire/issues/7046
